cli/src/stake.rs

```diff
@@ -1486,19 +1496,84 @@ pub fn process_deactivate_stake_account(
     fee_payer: SignerIndex,
 ) -> ProcessResult {
     let recent_blockhash = blockhash_query.get_blockhash(rpc_client, config.commitment)?;
-    let stake_authority = config.signers[stake_authority];
 
     let stake_account_address = if let Some(seed) = seed {
         Pubkey::create_with_seed(stake_account_pubkey, seed, &stake::program::id())?
     } else {
         *stake_account_pubkey
     };
 
-    let ixs = vec![stake_instruction::deactivate_stake(
-        &stake_account_address,
-        &stake_authority.pubkey(),
-    )]
+    let ixs = vec![if deactivate_delinquent {
```

Comment:
Gate all this business by `if !sign_only`? Might also be nice to make the `delinquent` cli arg conflict with `SIGN_ONLY_ARG.name`.